stores/filestore.go

```diff
@@ -441,25 +492,24 @@ var (
 
 // openFile opens the file specified by `filename`.
 // If the file exists, it checks that the version is supported.
-// If no file mode is provided, the file is created if not present,
+// The file is created if not present, opened in Read/Write and Append mode.
+var openFile = func(fileName string) (*os.File, error) {
+	return openFileWithFlags(fileName, defaultFileFlags)
+}
+
+// openFileWithModes opens the file specified by `filename`, using
+// the `modes` as open flags.
+// If the file exists, it checks that the version is supported.
+// If no open mode override is provided, the file is created if not present,
 // opened in Read/Write and Append mode.
-func openFile(fileName string, modes ...int) (*os.File, error) {
+func openFileWithFlags(fileName string, flags int) (*os.File, error) {
 	checkVersion := false
 
-	mode := os.O_RDWR | os.O_CREATE | os.O_APPEND
-	if len(modes) > 0 {
-		// Use the provided modes instead
-		mode = 0
-		for _, m := range modes {
-			mode |= m
-		}
-	}
-
 	// Check if file already exists
 	if s, err := os.Stat(fileName); s != nil && err == nil {
 		checkVersion = true
 	}
-	file, err := os.OpenFile(fileName, mode, 0666)
+	file, err := os.OpenFile(fileName, flags, 0666)
```

Comment:
Yep, flipped them around.  User = r/w, group = r, other = nothing.  A different PR, allowing the user to supply a mask sounds good here.